src/packages/v4/views/V4ProjectDashboard/V4ProjectTabs/CyclesPayoutsPanel/hooks/useV4FormatConfigurationTokenSection.ts

```diff
@@ -0,0 +1,218 @@
+import { t } from '@lingui/macro'
+import { flagPairToDatum } from 'components/Project/ProjectHeader/utils/flagPairToDatum'
+import { pairToDatum } from 'components/Project/ProjectHeader/utils/pairToDatum'
+import {
+  ConfigurationPanelDatum,
+  ConfigurationPanelTableData,
+} from 'components/Project/ProjectTabs/CyclesPayoutsTab/ConfigurationPanel'
+
+
+import { Ruleset } from 'packages/v4/models/ruleset'
+import { RulesetMetadata } from 'packages/v4/models/rulesetMetadata'
+import { useMemo } from 'react'
+import { tokenSymbolText } from 'utils/tokenSymbolText'
+
+export const useV4FormatConfigurationTokenSection = ({
+  ruleset,
+  rulesetMetadata,
+  tokenSymbol: tokenSymbolRaw,
+  queuedRuleset,
+  queuedRulesetMetadata,
+}: {
+  ruleset: Ruleset | undefined | null
+  rulesetMetadata: RulesetMetadata | undefined | null
+  tokenSymbol: string | undefined
+  queuedRuleset: Ruleset | undefined | null
+  queuedRulesetMetadata?: RulesetMetadata | undefined | null
+}): ConfigurationPanelTableData => {
+  const tokenSymbol = useMemo(
+    () =>
+      tokenSymbolText({
+        tokenSymbol: tokenSymbolRaw,
+        capitalize: false,
+        plural: true,
+      }),
+    [tokenSymbolRaw],
+  )
+  const decayRateFloat = ruleset?.decayRate.toFloat()
+  const currentTotalIssuanceRate = ruleset?.weight.toFloat()
+  const queuedTotalIssuanceRate = queuedRuleset ? 
```

Comment:
This hook feels weird to me. Why is it combining the queued ruleset and current ruleset. Could we just have a hook that takes a ruleset, does the formatting, and we just pass in the current or queued depending on the selection?